source/draw/gpu/GpuCanvas.ooc

```diff
@@ -72,7 +76,9 @@ GpuCanvasYuv420Semiplanar: class extends GpuSurface {
 		this _target y canvas fill()
 		this _target uv canvas fill()
 	}
-	draw: override func ~mesh (image: Image, mesh: GpuMesh) {
+	draw: override func ~mesh (image: GpuImage, mesh: GpuMesh) {
+		if (!image instanceOf?(GpuYuv420Semiplanar))
+			Debug raise("Invalid input format in GpuYuv420SemiplanarCanvas::draw~mesh")
```

Comment:
Apart from changing the argument parameter, I don't think so, as the types are meant to be resolved at runtime.